algorithms/src/Kokkos_Sort.hpp

```diff
@@ -628,10 +613,23 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
   bin_sort.sort(exec, view);
 }
 
+template <class ExecutionSpace, class DataType, class... Properties>
+std::enable_if_t<(Kokkos::is_execution_space<ExecutionSpace>::value) &&
+                 (SpaceAccessibility<
+                     HostSpace, typename Kokkos::View<DataType, Properties...>::
+                                    memory_space>::accessible)>
+sort(const ExecutionSpace& space,
+     const Kokkos::View<DataType, Properties...>& view) {
+  space.fence("Kokkos::sort: Fence before sorting on the host");
+  auto first = Experimental::begin(view);
+  auto last  = Experimental::end(view);
+  std::sort(first, last);
+}
+
 template <class ViewType>
-void sort(ViewType const& view, bool const always_use_kokkos_sort = false) {
+void sort(ViewType const& view) {
   typename ViewType::execution_space exec;
-  sort(exec, view, always_use_kokkos_sort);
+  sort(exec, view);
```

Comment:
I think this should have a global `fence` before calling the overload taking an execution space but that might be debatable.

Reply:
why?

Reply:
Generally, we expect functions without execution space argument to apply all possibly necessary fences and the input for `sort` might depend on the completion of another kernel (for which we don't know the execution space instance).

Reply:
but this one here accepts an execution space instance

Reply:
No, this function is `void sort(ViewType const& view)` (without execution space instance) and we create an execution space instance to call the overload that takes an execution space instance (which should not have any fences).

Reply:
ah you are right, sorry I misread

Reply:
Added.